qiime_studio/api/v1.py

```diff
@@ -50,4 +57,61 @@ def api_workflows(plugin_name):
             {'name': name, 'type': repr(type_[0])}
             for name, type_ in value.signature.outputs.items()
         ]
-    return jsonify({"workflows": workflows_dict})
+    return jsonify({'workflows': workflows_dict})
+
+
+@v1.route('/artifacts', methods=['GET'])
+def api_artifacts():
+    artifact_paths = glob.glob(os.path.join(os.getcwd(), '*.qzf'))
+    artifacts = [
+        {
+            'name': os.path.splitext(os.path.split(path)[1])[0],
+            'uuid': str(Artifact.load(path).uuid),
+            'type': str(Artifact.load(path).type),
+            'path': path,
+            'uri': 'artifacts/%s' % (os.path.splitext(
+                                        os.path.split(path)[1])[0])
+        }
+        for path in artifact_paths
+    ]
+    return jsonify({'artifacts': artifacts})
+
+
+@v1.route('/artifacts/<name>', methods=['DELETE'])
+def delete_artifact(name):
+    result = {
+        'success': True
+    }
+    artifact_json = request.get_json()['artifact']
+    artifact = Artifact.load(artifact_json['path'])
+    if str(artifact.uuid) == artifact_json['uuid']:
+        try:
+            os.remove(artifact_json['path'])
+        except OSError:
+            result['success'] = False
+
+    return jsonify(result)
+
+
+@v1.route('/artifacts/<plugin_name>/<workflow_name>/<input_name>', methods=['GET'])
+def api_input_artifacts(plugin_name, workflow_name, input_name):
+    plugin = PLUGIN_MANAGER.plugins[plugin_name]
+    workflow = plugin.workflows[workflow_name]
+    input_type = workflow.signature.inputs[input_name][0]
+    input_artifacts = []
+    artifact_paths = glob.glob(os.path.join(os.getcwd(), '*.qzf'))
+    artifacts = [
+        {
+            'name': os.path.splitext(os.path.split(path)[1])[0],
+            'uuid': str(Artifact.load(path).uuid),
+            'type': str(Artifact.load(path).type),
+            'path': path,
+            'uri': 'artifacts/%s' % (os.path.splitext(
+                                        os.path.split(path)[1])[0])
+        }
+        for path in artifact_paths
+    ]
+    for artifact in artifacts:
+        if Artifact.load(artifact['path']).type <= input_type:
+            input_artifacts.append(artifact)
```

Comment:
Oh nice, I was trying to think of how I could cut down on all the loading as it's even happening twice per Artifact in the original list of dicts generation. Would you have to do:

```python
for idx, artifact in enumerate(map(Artifact.load, artifact_paths)):
    ...
```
So that you can get the path? I see that some internal methods of Artifacts can get their path, but didn't see a public way to get at it, so would we need to be able to use 
```python
'path': artifact_paths[idx]
``` 
in the JSON portion to fully build our JSON Artifacts?